packages/vite/src/node/config.ts

```diff
@@ -294,8 +302,12 @@ export async function resolveConfig(
     [`package.json`],
     true /* pathOnly */
   )
-  const optimizeCacheDir =
-    pkgPath && path.join(path.dirname(pkgPath), `node_modules/${DEP_CACHE_DIR}`)
+  let cacheDir: ResolvedConfig['cacheDir']
+  if (config.cacheDir) {
+    cacheDir = path.resolve(resolvedRoot, config.cacheDir)
+  } else {
+    cacheDir = pkgPath && path.join(path.dirname(pkgPath), `node_modules/.vite`)
+  }
```

Comment:
**suggestion (non-blocking):** What about something like this?
_May be correctly formatted in IDE_

```suggestion
  const cacheDir: ResolvedConfig['cacheDir'] = !!config.cacheDir
    ? path.resolve(resolvedRoot, config.cacheDir)
    : pkgPath && path.join(path.dirname(pkgPath), `node_modules/.vite`)
```

Reply:
I've considered about this, but I finally thought it's less readable because of the `pkgPath &&` part 🤔 

I'll try to tweak here